src/VisualStudio/Xaml/Impl/Implementation/XamlProjectService.cs

```diff
@@ -271,4 +275,9 @@ private void OnDocumentMonikerChanged(uint docCookie, IVsHierarchy hierarchy, st
 
         return null;
     }
+
+    public void Dispose()
+    {
+        _documentClosedHandlerDisposer.Dispose();
+    }
```

Comment:
my preference (applies to all files where  you added this) is that Dispose be right after the constructor.  taht way you can see the creation/cleanup right next to each other.